### PR TITLE
User registration & log in

### PR DESCRIPTION
**Pull request by CC-3636**
Serialization working, tests added, and refactoring

### CODE REVIEW

File: src/main/java/data/courses.csv [hunk 1]
```diff
@@ -0,0 +1 @@
+course_name,course_instructor,tasks?
```
[the remark this answers is not in the document]
Comment (CC-3636) [reply src/main/java/data/courses.csv:1]: Thanks for the feedback, Tiare! Implemented all of the changes you requested to the InstructorMain and also renamed MainScreen to StudentMainScreen. Moved users.ser and userstest.ser to data packages.

**Review by TiareMar — APPROVED** [review]

**Review by yyutiw — DISMISSED** [review]
Code style: overall good job, methods are descriptive, variable names are as well (for the most part, some suspect naming in test files lol)
Clean arch: don't see any problems, no blatant violations of SOLID
Documentation: javadoc comments throughout make code really easy to follow along

good job :)

**Review by jltng — COMMENTED** [review]
The file wasn't in this PR but for StudentUser.java, the addCourse method was adding the course to student.todolist instead of student.course. The correct method should be as follows:
Screen Shot 2022-12-02 at 2 40 35 PM

File: src/main/java/screens/login_registration/FileUser.java [hunk 2]
```diff
@@ -75,13 +94,4 @@ public Map<String, UserRegSaveRequest> getAccounts() {
         return this.accounts;
```
Comment (jltng) [inline src/main/java/screens/login_registration/FileUser.java:94]: For my enrolment use case, I use this method to access the students in the map and then search for the desired student. To do so I would need this to be a static method:
Screen Shot 2022-12-02 at 2 42 47 PM

File: src/main/java/screens/login_registration/LogoutResponseFormatter.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    /**
+     * @param logout the logout response
+     * @return the successful logout view
+     */
+    @Override
+    public LogoutResponseModel prepareSuccessView(LogoutResponseModel logout) {
+        return logout;
+    }
+
+}
```
Comment (jltng) [inline src/main/java/screens/login_registration/LogoutResponseFormatter.java:17]: It doesn't really matter in terms of functionality, but this should be the Presenter (and the presenter file in your use case should be the OutputBoundary

**Review by jltng — CHANGES_REQUESTED** [review]

**Review by jltng — DISMISSED** [review]
Approved because there are no major bugs that break the program, and all other features build on this

File: src/main/java/screens/login_registration/FileUser.java (hunk 2 above)
Comment (CC-3636) [reply src/main/java/screens/login_registration/FileUser.java:94]: You can call FileUser from your interactor bc of the dependency inversion--FIleUser implements UserGateway and LoginGateway and LogoutGateway which are in the use case layer

**Review by jltng — APPROVED** [review]
Requested changes addressed. Looks good!

**Review by yyutiw — APPROVED** [review]

**Review by TiareMar — APPROVED** [review]
Approving because I see no significant issues

**Pull request merged**